CREATE2 opcode support

mersinvald opened the pull request:
Add support for CREATE2 opcode:
 - separate TransactionAction
 - Address generation according to https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1014.md

Commit:
create2 transaction action

Commit:
accept code by Rc link

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ETCDEVTeam/etcommon-rs?pullRequest=67) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ETCDEVTeam/etcommon-rs?pullRequest=67) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ETCDEVTeam/etcommon-rs?pullRequest=67) it.</sub>

Commit:
no_std support

Commit:
version bump

r8d8 reviewed (approved):
LGTM;

tzdybal commented on block-core/src/transaction.rs:
```diff
@@ -34,16 +48,60 @@ impl Encodable for TransactionAction {
             },
+            &TransactionAction::Create2(salt, ref code) => {
+                s.begin_list(3)
+                    .append(&0xc2_u8)
```
I don't get the code (here and below) related to `0xc2_u8` and I don't see this value anywhere in the EIP-1014 - maybe introducing properly named constant would make code more readable?

mersinvald replied:
That's a descriptor value, I've added it in case we'd need to add more enum variants in TransactionAction. 

Withoug it it would be impossible to add more variants in the future without breaking compatibility for the original Call and Create.

mersinvald replied:
Though, is TransactionAction encoding standardized?

mersinvald commented:
@tzdybal fixed the magic number. Can you merge PR please? I don't have rights somehow.

mersinvald commented:
Noticed that I initially bumped a minor version though removal of Copy derive introduces breaking changes. Fixed that.

Also fixed the magic number @tzdybal.

Can you merge it please? I somehow have no rights to do that in this repo.

tzdybal commented:
I'm not able to merge. Maybe @r8d8?

Commit:
unmagify magic number

Commit:
major version bump sue to breaking change

Commit:
bump version in deps

tzdybal commented on block-core/src/transaction.rs:
```diff
@@ -21,10 +26,21 @@ impl TransactionAction {
+                digest.input(&[0xff]);
+                digest.input(&caller);
+                digest.input(&salt);
+                digest.input(code);
```
According to https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1014.md the hash of code should be used, not the actual value.

whilei replied:
https://github.com/ethereum/EIPs/pull/1014/files#r219843015

https://github.com/ethereum/EIPs/pull/1375

Commit:
bump major version due to switching to block-core 0.2

Commit:
use code hash for address calculation

mersinvald commented:
@tzdybal @r8d8 fixed, also bumped a version of block crate.

tzdybal reviewed (approved):
LGTM
